arbnode/batch_poster.go

```diff
@@ -208,6 +212,84 @@ func NewBatchPoster(l1Reader *headerreader.HeaderReader, inbox *InboxTracker, st
 	return b, nil
 }
 
+// checkRevert checks blocks with number in range [from, to] whether they
+// contain reverted batch_poster transaction.
+func (b *BatchPoster) checkReverts(ctx context.Context, from, to *big.Int) (bool, error) {
+	if from.Cmp(to) == 1 {
+		return false, fmt.Errorf("wrong range, from: %v is more to: %v", from.Int64(), to.Int64())
+	}
+	one := big.NewInt(1)
+	for number := new(big.Int).Set(from); number.Cmp(to) != 1; number.Add(number, one) {
+		block, err := b.l1Reader.Client().BlockByNumber(ctx, number)
+		if err != nil {
+			return false, fmt.Errorf("getting block: %v by number: %w", number.Int64(), err)
+		}
+		for idx, tx := range block.Transactions() {
+			from, err := b.l1Reader.Client().TransactionSender(ctx, tx, block.Hash(), uint(idx))
+			if err != nil {
+				return false, fmt.Errorf("getting sender of transaction tx: %v, %w", tx.Hash(), err)
+			}
+			if bytes.Equal(from.Bytes(), b.dataPoster.From().Bytes()) {
+				r, err := b.l1Reader.Client().TransactionReceipt(ctx, tx.Hash())
+				if err != nil {
+					return false, fmt.Errorf("getting a receipt for transaction: %v, %w", tx.Hash(), err)
+				}
+				if r.Status == types.ReceiptStatusFailed {
+					return true, nil
+				}
+			}
+		}
+	}
+	return false, nil
+}
+
+// pollForReverts runs a gouroutine that listens to l1 block headers, checks
+// if any transaction made by batch poster was reverted.
+func (b *BatchPoster) pollForReverts(ctx context.Context) {
+	headerCh, unsubscribe := b.l1Reader.Subscribe(false)
+	defer unsubscribe()
+	var (
+		last *big.Int // number of last seen block
+		one  = big.NewInt(1)
```

Comment:
There's `common.Big1` which you can use instead

Reply:
There's a lot of big math here that gets complicated. It's safe to assume block numbers are uint64s, so if it makes the code cleaner we could just immediately turn the block numbers into uint64s and do math that way.

Reply:
Done. Used int64 instead of uint64 though.